tests/ui/macros/std-2024-macros.rs

```diff
@@ -0,0 +1,13 @@
+// Tests a small handful of macros in the standard library how they handle the
+// new behavior introduced in 2024.
```

Comment:
Suggestion: maybe slightly elaborate what said "new behavior" is?

Reply:
Sure!